### ISSUE TITLE
can not reduce model reduce

### ISSUE THREAD

**Issue by kikyou123** [coxlab/prednet#31]
when i download your model, the Model MSE is 0.08421. 
But when I run the kitti_train.py and use the trained weights, I get the Model MSE 0.018259
this is my results in the test set:
![image](https://user-images.githubusercontent.com/19486038/30911862-d6d6140e-a3bc-11e7-8afd-19b77dd2c6f7.png)
Is the model in download_models.sh using the kitti_train.py?

**Comment by bill-lotter** [comment]
Yes the weights are trained using kitti_train.  With the downloaded weights, are you missing a zero and it's 0.008421?  Because that's about what I get when I use the tensorflow backend (with theano, I get ~0.007).  Not sure why it seems to not train as well for some people.  Easy first things to try are different optimizers/learning rates/momentum.  Sometimes Rmsprop works better when adam struggles, so you can give that a try.

**Comment by robbiewu008** [comment]
@kikyou123 I got the similar result with you, my env is tensorflow-gpu 1.2.1, keras 2.0.6 with cudnn v5.1 for cuda 8.0.
----------------------
Model MSE: 0.017124
Previous Frame MSE: 0.021246

**Comment by kikyou123** [comment]
@robbiewu008  It seems there is a bug in the code.
`rrors_by_time = TimeDistributed(Dense(1, weights=[layer_loss_weights, np.zeros(1)], trainable=False), trainable=False)(errors)  # calculate weighted error by layerr`

**Comment by robbiewu008** [comment]
@kikyou123 missing the Reshape, could you share it?
![image](https://user-images.githubusercontent.com/4103621/33198205-aabd39ca-d124-11e7-8282-59eb92a0779e.png)

**Comment by kikyou123** [comment]
@robbiewu008 
![image](https://user-images.githubusercontent.com/19486038/33213082-f29a8fd6-d160-11e7-98dc-cd46e71d6c28.png)

**Comment by robbiewu008** [comment]
@kikyou123 still don't work.

![image](https://user-images.githubusercontent.com/4103621/33226491-f7a3040e-d1c9-11e7-8577-795b5b34295f.png)

**Comment by kikyou123** [comment]
sorry, It is multi prediction code. You can change 
![image](https://user-images.githubusercontent.com/19486038/33226966-3291f2c8-d1d4-11e7-99ad-a598ab989edb.png)
15 to 10

**Comment by robbiewu008** [comment]
@kikyou123 It works now, thank you so much!

Model MSE: 0.006937
Previous Frame MSE: 0.021246

**Comment by flyingmoth** [comment]
@kikyou123  Did you successfully train the model for muti-step prediction?  I get the stable frames over time step.

**Comment by bill-lotter** [comment]
@kikyou123 It looks like there's an issue with TimeDistributed in Keras 2 where it overrides the initial weights of the layer to be wrapped (https://github.com/keras-team/keras/issues/8895). I changed the kitti_train.py code to account for this (https://github.com/coxlab/prednet/commit/9f6482e6e6038e02aa3dbd878bae23c7b3a5a28a). It should work now, but let me know if you find any other issues - thanks!

**Comment by alvinxiii** [comment]
@kikyou123 @bill-lotter  hi. I tried run the multi prediction code. I barely see the differences from it. Is this the correct steps?
This is my step:
1. run kitti_train.py
2. run kitti_extrap_finetune.py
3. run kitti_evaluate.py
